pycqed/instrument_drivers/meta_instrument/device.py

```diff
@@ -0,0 +1,1012 @@
+            fsqs.chevron_seqs(
+                qbc_name=qbc.name, qbt_name=qbt.name, qbr_name=qbr.name,
+                hard_sweep_dict=hard_sweep_params,
+                soft_sweep_dict=soft_sweep_params,
+                operation_dict=operation_dict,
+                cz_pulse_name=cz_pulse_name,
+                num_cz_gates=num_cz_gates,
+                cal_points=cp, upload=False, prep_params=prep_params)
+
+        if return_seq:
+            return sequences
+
+        hard_sweep_func = awg_swf.SegmentHardSweep(
+            sequence=sequences[0], upload=upload,
+            parameter_name=list(hard_sweep_params)[0],
+            unit=list(hard_sweep_params.values())[0]['unit'])
+        MC.set_sweep_function(hard_sweep_func)
+        MC.set_sweep_points(hard_sweep_points)
+
+        # sweep over flux pulse amplitude of qbc
+        channels_to_upload = [qbc.flux_pulse_channel()]
+        MC.set_sweep_function_2D(awg_swf.SegmentSoftSweep(
+            hard_sweep_func, sequences,
+            list(soft_sweep_params)[0], list(soft_sweep_params.values())[0]['unit'],
+            channels_to_upload=channels_to_upload))
+        MC.set_sweep_points_2D(soft_sweep_points)
+        MC.set_detector_function(qbr.int_avg_classif_det if classified
+                                 else qbr.int_avg_det)
+        if exp_metadata is None:
+            exp_metadata = {}
+        exp_metadata.update({'preparation_params': prep_params,
+                             'cal_points': repr(cp),
+                             'rotate': len(cal_states) != 0,
+                             'data_to_fit': {qbr.name: 'pe'},
+                             'hard_sweep_params': hard_sweep_params,
+                             'soft_sweep_params': soft_sweep_params})
+        MC.run_2D(name=label, exp_metadata=exp_metadata)
+
+        if analyze:
+            tda.MultiQubit_TimeDomain_Analysis(qb_names=[qbr.name],
+                                               options_dict={'TwoD': True})
+
+    def measure_cphase(self, qbc, qbt, soft_sweep_params, cz_pulse_name, prep_params=None,
+                       label=None, cal_states='auto', n_cal_points_per_state=1,
+                       num_cz_gates=1, hard_sweep_params=None, exp_metadata=None,
+                       analyze=True, upload=True, for_ef=True, max_flux_length=None, **kw):
+        '''
+        method to measure the leakage and the phase acquired during a flux pulse
+        conditioned on the state of the control qubit (self).
+        In this measurement, the phase from two Ramsey type measurements
+        on qb_target is measured, once with the control qubit in the excited state
+        and once in the ground state. The conditional phase is calculated as the
+        difference.
+
+        Args:
+            qbc (QuDev_transmon): control qubit / fluxed qubit
+            qbt (QuDev_transmon): target qubit / non-fluxed qubit
+        '''
+        qbc_name = qbc.name
+        qbt_name = qbt.name
```

Comment:
delete these two unused variables

Reply:
Addressed in https://github.com/QudevETH/PycQED_py3/pull/62/commits/90cdd02cd058de89758a207786f37740edd93bbf